viewing_party/party.py

```diff
@@ -1,23 +1,263 @@
+    for movie in user_data["watched"]:
+        genre=movie["genre"]
+        if genre in frequentcy_of_genre:
+            frequentcy_of_genre[genre]+=1
+        else:
+            frequentcy_of_genre[genre]=1
+
+    for frequency in frequentcy_of_genre:
+        if frequentcy_of_genre[frequency]  >  highest_friqency:
+            highest_friqency = frequentcy_of_genre[frequency] 
+            most_frequent_genre_name=frequency
+
+    return most_frequent_genre_name
+
+    
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+#Consider the movies that the user has watched, and consider the movies that their friends have watched. 
+# Determine which movies the user has watched, but none of their friends have watched.
+def get_unique_watched(user_data):
+    set_friends_watched=set()
+    list_friend=user_data["friends"]
+    for friend in list_friend:
+        moview_watched_friend=friend["watched"]
+        for movie in moview_watched_friend:
+            title=movie["title"]
+            set_friends_watched.add(title)
+    user_set_of_watched_movies=set()        
+    for movie in user_data["watched"]:
+        title_of_movie=movie['title']
+        user_set_of_watched_movies.add(title_of_movie)
+
+    unique_movies_set=user_set_of_watched_movies.difference(set_friends_watched) 
+    list_of_unique_movies=[]   
+    for movie in user_data["watched"]:
+        if movie["title"]in unique_movies_set:
+            list_of_unique_movies.append(movie)
+    return list_of_unique_movies  
+
+
+
+#Consider the movies that the user has watched, and consider the movies that their friends have watched. 
+# Determine which movies at least one of the user's friends have watched, but the user has not watched.
+def get_friends_unique_watched(user_data):
+    allMovies=[]
+    friends_watched=user_data["friends"]
+    for f in friends_watched:
+       
+        allMovies.extend(f["watched"])
+    
+    uniqueMovies = {}
+    for movie in allMovies:
+        movieTitle = movie["title"]
+        if movieTitle in uniqueMovies:
+            continue
+        else:
+            uniqueMovies[movieTitle] = movie
+    
+    print(uniqueMovies.values())
+    return uniqueMovies.values() 
+
+
+    #helper function
+def find_unique_user_movies(user_data):
+    list_of_titles=[]
+    watched =user_data["watched"]
+    for movie in watched:
+        title = movie["title"]
+        list_of_titles.append(title)
+    set_user_unique_movies=set(list_of_titles)   
+    return set_user_unique_movies
+
+
+def find_user_movies(user_data):
+    watched =user_data["watched"]
+    return watched
+
+
+def get_friends_unique_watched(user_data):
+    list_of_titles=[]
+    list_all_movies=[]
+    all_friends=user_data["friends"]
+
+    for f in all_friends:
+        friend_unique_movies = find_unique_user_movies(f)
+        list_of_titles.extend(friend_unique_movies)

```

Comment:
The function `find_unique_user_movies` returns a set, and the end goal is a set of unique titles.  I recommend looking into how to combine sets rather than building a list and converting it to a set: https://www.pythonpool.com/learn-how-to-combine-sets-in-python/